python/tank/platform/validation.py

```diff
@@ -379,16 +389,31 @@ def validate(self, settings):
         # first sanity check that the schema is correct
         validate_schema(self._display_name, self._schema)
         
-        # Ensure that all required keys are in the settings and that the
-        # values are appropriate.
+        # Ensure that all keys are in the settings or have a default value in
+        # the manifest. Also make sure values are appropriate.
         for settings_key in self._schema:
             value_schema = self._schema.get(settings_key, {})
-            
-            # make sure the required key exists in the environment settings
-            if settings_key not in settings:
-                raise TankError("Missing required key '%s' in settings!" % settings_key)
-            
-            self.__validate_settings_value(settings_key, value_schema, settings[settings_key])
+
+            if settings_key in settings:
+                # value exists in the settings. use it.
+                settings_value = settings[settings_key]
+            else:
+                # No value in the settings. Look for a default value.
+                if value_schema.get("type") == "hook":
+                    # Hooks already have the concept of a default value. If this
+                    # setting is a hook, just use that special default hook
+                    # string and allow the validation code do its thing.
+                    settings_value = constants.TANK_BUNDLE_DEFAULT_HOOK_SETTING
```

Comment:
Examples of this:

### old style
- manifest definition: https://github.com/shotgunsoftware/tk-multi-publish/blob/master/info.yml#L65
- installed in config: https://github.com/shotgunsoftware/tk-config-default/blob/master/env/shot_step.yml#L71
So here, tk will discover *default* look at the manifest value (`primary_publish`) and resolve the hook path `PATH_TO_BUNDLE/hooks/primary_publish.py` at runtime.

### new style
- manifest def: https://github.com/shotgunsoftware/tk-multi-loader2/blob/master/info.yml#L31
- in config: https://github.com/shotgunsoftware/tk-config-default/blob/master/env/shot_step.yml#L479

Here, we don't need to do the funny roundtrip into the manifest that we did with the old style hooks (e.g. to figure out that `default` actually means `primary_publish` in the case of the setting above). Instead, it's just baked out and much more readable in the config: `actions_hook: '{self}/tk-maya_actions.py'` - it is easy to see what tk will be doing (good transparency) and it's easy to configure, typically a good starting point it so just replace `{self}` with `{config}`.

Also note how in this case, the manifest def `default_value: "{self}/{engine_name}_actions.py"` has got `{engine_name}` resolved at install time but the rest of the tokens resolved at runtime.